src/redux/blockchain/blockchainActions.js

```diff
@@ -78,10 +80,14 @@ export const connect = () => {
           });
           ethereum.on("chainChanged", () => {
             window.location.reload();
-          });
+          })
           // Add listeners end
         } else {
-          dispatch(connectFailed(`Change network to ${CONFIG.NETWORK.NAME}.`));
```

Comment:
Instead of just putting this in the else clause, the if/else block should be removed completely and the `wallet_switchEthereumChain` request can be made after line 64 and will not require the user to interact multiple times with the connect button.

Reply:
How we gonna check networkId if we remove the if/else block? I didn't understand exactly may you describe little bit more.

Reply:
I made a branch from my fork to show you what I am talking about.

https://github.com/mqschwanda/hashlips_nft_minting_dapp/blob/5a65ab2adb2409a67c27fd1291d0e4c9b849403a/src/redux/blockchain/blockchainActions.js#L63-L68

The code I have linked will attempt to switch the Ethereum chain if the metamask wallet is not connected to the network. If the switch is complete the function will continue on with the rest of the connecting process. This is all done with one click of the connect button by the user.

In your code, if the user is on the wrong network they will be prompted to switch chains, but after they switch they will be required to click the connect button again which is not the ideal user experience.

So it really comes down to user experience and efficiency. One-click is going to be better compared to two clicks. This may appear like a minor issue but when dealing with users, eliminating confusion and complexity is the name of the game.

Reply:
I thought I should remove the second click requirement, but now I'm surprised why I didn't notice such a simple thing. Thanks a thousand times.

Reply:
No problem, and nice work on the code, hopefully we can get it merged.